frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala

```diff
@@ -536,35 +622,62 @@ class BspProtocolSpec {
               val json = taskFinish.data.get
               bsp.CompileReport.decodeCompileReport(json.hcursor) match {
                 case Right(report) =>
+
                   if (isMainProject(report.target)) {
                     if (receivedReports == 0) {
-                      // This is the batch compilation which should have a warning and no errors
+                      val received = findDiagnosticsReport(report.target, receivedReports)
+                      val expected =
+                        "TextDocumentIdentifier(Uri(file:/Users/jvican/Code/bloop/frontend/src/test/resources/cross-test-build-0.6/test-project/shared/src/main/scala/hello/App.scala)) List(Diagnostic(Range(Position(5,8),Position(5,8)),Some(Warning),None,None,local val in method main is never used,None)) (reset = true)"
+                      Assert.assertEquals(expected, received)
                       receivedReports += 1
-                      Assert.assertEquals(s"Warnings in $MainProject != 1", 1, report.warnings)
-                      Assert.assertEquals(s"Errors in $MainProject != 0", 0, report.errors)
-                      //Assert.assertTrue(s"Duration in $MainProject == 0", report.time != 0)
                     } else if (receivedReports == 1) {
-                      // This is the incremental compile which should have errors
+                      // Check that we only receive one error
+                      val received = findDiagnosticsReport(report.target, receivedReports)
+                      // We don't expect a warning here because the client is supposed to reuse the previous diagnostic (we *didn't* send a notification for that file with reset=true)
+                      val expected =
+                        "TextDocumentIdentifier(Uri(file:/Users/jvican/Code/bloop/frontend/src/test/resources/cross-test-build-0.6/test-project/jvm/src/main/scala/TestIncrementalCompilation.scala)) List(Diagnostic(Range(Position(3,13),Position(3,13)),Some(Error),None,None,']' expected but ')' found.,None)) (reset = true)"
+                      Assert.assertEquals(expected, received)
                       receivedReports += 1
-                      Assert.assertTrue(
-                        s"Expected errors in incremental cycle of $MainProject",
-                        report.errors > 0
+                    } else if (receivedReports == 2) {
+                      // The following operation is a no-op, so we should get an empty diagnostic with reset enabled to delete the previous error
+                      val received = findDiagnosticsReport(report.target, receivedReports)
+                      val expected =
+                        "TextDocumentIdentifier(Uri(file:/Users/jvican/Code/bloop/frontend/src/test/resources/cross-test-build-0.6/test-project/jvm/src/main/scala/TestIncrementalCompilation.scala)) List() (reset = true)"
```

Comment:
URIs should be relativized by the workspace root directory.

Reply:
What's your rationale for relativizing? Saving network bandwidth? In general, I prefer to always use absolute paths over relative paths

Reply:
Uri should stay absolute in the server, comment was only to make it relative in this expect test.